### ISSUE TITLE
Function Refactor for 4.0

### ISSUE THREAD

**Issue by HamptonMakes** [sass/sass#1903]
My thoughts are that we should take the functions as they are, especially the prefixed ones, and in prep for `4.0`, we can reorganize stuff so that it makes more sense and will make it so that @nex3 can focus on the important parts of the module system.

My proposal is as follows:
- Break up the functions into the groups that will (eventually) become modules (str-, map-)
- Create prefixed versions of all (even legacy) functions and use those as the current recommended
- Allow for simple deprecation warnings of "str-" prefixed (and non-prefixed legacy) functions in `4.0` itself
- Change the way we use `declare` to make deprecation warnings easier, and to build a logical grouping around what will become standard lib modules later.

Basically, @nex3, would you be up for letting me dig into the functions and function tests and start getting them ready for the work that you'll be doing with 4.0?

**Comment by HamptonMakes** [comment]
End the tyranny of a 2,500+ line file! ;)

**Comment by nex3** [comment]
We're still a ways out from even having a clear idea of what the module system will look like. It seems premature to start modifying infrastructure and recommended usage around a system we haven't begun to nail down.

Even once we do have it, I don't think the infrastructure of moving functions into modules will be particularly difficult. There's not a whole lot of benefit to be had from moving eagerly on this, and there are a lot of drawbacks.

**Comment by HamptonMakes** [comment]
Just trying to find a way to pitch in since you were asking for help with the Ruby implementation, and a 2,000+ line file is something that kind of sticks out as needing a refactor. One day a week isn't a ton to get all this done, so was trying to pitch in.

Sounds like only after 4.0 are people able to pitch in on this stuff.

**Comment by nex3** [comment]
> Sounds like only after 4.0 are people able to pitch in on this stuff.

I guess that depends what you mean by "this stuff"? There are [plenty of issues](https://github.com/sass/sass/labels/Help%20Wanted) that are explicitly marked as good for contributions.
